src/env-inl.h

```diff
@@ -370,23 +330,26 @@ inline void Environment::set_trace_sync_io(bool value) {
   trace_sync_io_ = value;
 }
 
-inline uint32_t* Environment::heap_statistics_buffer() const {
-  CHECK_NE(heap_statistics_buffer_, nullptr);
-  return heap_statistics_buffer_;
+inline uint32_t* Environment::heap_statistics_buffer() {
+  CHECK_NE(heap_statistics_buffer_.size(), 0);
+  return &heap_statistics_buffer_.front();
 }
 
-inline void Environment::set_heap_statistics_buffer(uint32_t* pointer) {
-  CHECK_EQ(heap_statistics_buffer_, nullptr);  // Should be set only once.
-  heap_statistics_buffer_ = pointer;
+inline void Environment::set_heap_statistics_buffer_length(size_t length) {
+  CHECK_EQ(heap_statistics_buffer_.size(), 0);  // Should be set only once.
+  heap_statistics_buffer_.resize(length);
 }
 
-inline char* Environment::http_parser_buffer() const {
-  return http_parser_buffer_;
+inline char* Environment::http_parser_buffer() {
+  if (http_parser_buffer_.size() == 0)
+    return nullptr;
+  else
+    return &http_parser_buffer_.front();
 }
 
-inline void Environment::set_http_parser_buffer(char* buffer) {
-  CHECK_EQ(http_parser_buffer_, nullptr);  // Should be set only once.
-  http_parser_buffer_ = buffer;
+inline void Environment::set_http_parser_buffer_length(size_t length) {
+  CHECK_EQ(http_parser_buffer_.size(), 0);  // Should be set only once.
+  http_parser_buffer_.resize(length);
 }
```

Comment:
Can you replace the `vector.size() == 0` checks with `vector.empty()` and `&vector.front()` with `&vector[0]`?